### PR TITLE
Rename global routing to endpoint routing

### PR DESCRIPTION
**Pull request by JamesNK**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs [hunk 1]
```diff
@@ -64,15 +64,22 @@ public MvcOptions()
...
 
-        // REVIEW: Add documentation
+        [Obsolete("Use EnableEndpointRouting.")]
         public bool EnableGlobalRouting
         {
-            get => _enableGlobalRouting.Value;
-            set => _enableGlobalRouting.Value = value;
+            get => EnableEndpointRouting;
+            set => EnableEndpointRouting = value;
+        }
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs]: It should be possible to just delete this right? I don't think it leaked out of this repo...

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/CorsDispatchingTests.cs [hunk 2]
```diff
@@ -9,9 +9,9 @@
 
 namespace Microsoft.AspNetCore.Mvc.FunctionalTests
 {
-    public class CorsGlobalRoutingTests : CorsTestsBase<CorsWebSite.StartupWithGlobalRouting>
+    public class CorsEndpointRoutingTests : CorsTestsBase<CorsWebSite.StartupWithEndpointRouting>
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/CorsDispatchingTests.cs]: This filename still has `Dispatch`in in the name.

File: src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs (hunk 1 above)
Comment (JamesNK) [reply src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs]: Web hooks used it to get their tests passing

Comment (rynowak) [reply src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs]: ah gotcha!

**Pull request merged**